src/SqlPersistence.PersistenceTests/PersistenceTestsConfiguration.cs

```diff
@@ -1,174 +1,207 @@
-
-        public Task Configure(CancellationToken cancellationToken = default)
+        var postgresConnectionString = Environment.GetEnvironmentVariable("PostgreSqlConnectionString");
+        if (!string.IsNullOrWhiteSpace(postgresConnectionString))
         {
-            var variant = (SqlTestVariant)Variant.Values[0];
-            var dialect = variant.Dialect;
-            var buildDialect = variant.BuildDialect;
-            var connectionFactory = () => variant.Open();
-            var pessimisticMode = variant.UsePessimisticMode;
-
-            if (SessionTimeout.HasValue)
-            {
-                dialect = new TimeoutSettingDialect(dialect, (int)SessionTimeout.Value.TotalSeconds);
-            }
+            RegisterCommonVariants(variants, new SqlDialect.PostgreSql(), BuildSqlDialect.PostgreSql);
 
-            var infoCache = new SagaInfoCache(
-                null,
-                Serializer.JsonSerializer,
-                reader => new JsonTextReader(reader),
-                writer => new JsonTextWriter(writer),
-                "PersistenceTests_",
-                dialect,
-                SagaMetadataCollection,
-                name => ShortenSagaName(name));
-
-            var connectionManager = new ConnectionManager(connectionFactory);
-            SagaIdGenerator = new DefaultSagaIdGenerator();
-            SagaStorage = new SagaPersister(infoCache, dialect);
-            OutboxStorage = CreateOutboxPersister(connectionManager, dialect, false, false);
-            SupportsPessimisticConcurrency = pessimisticMode;
-            CreateStorageSession = () => new StorageSession(connectionManager, infoCache, dialect);
-
-            GetContextBagForSagaStorage = () =>
-            {
-                var contextBag = new ContextBag();
-                contextBag.Set(new IncomingMessage("MessageId", [], Array.Empty<byte>()));
-                return contextBag;
-            };
+            variants.Add(CreateVariant(new SqlDialect.PostgreSql(),
+                BuildSqlDialect.PostgreSql,
+                usePessimisticModeForOutbox: false,
+                isolationLevel: IsolationLevel.Snapshot));
+        }
 
-            GetContextBagForOutbox = () =>
-            {
-                var contextBag = new ContextBag();
-                contextBag.Set(new IncomingMessage("MessageId", [], Array.Empty<byte>()));
-                return contextBag;
-            };
+        if (!string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("MySQLConnectionString")))
+        {
+            RegisterCommonVariants(variants, new SqlDialect.MySql(), BuildSqlDialect.MySql);
+        }
 
-            using (var connection = connectionFactory())
-            {
-                connection.Open();
+        if (!string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("OracleConnectionString")))
+        {
+            RegisterCommonVariants(variants, new SqlDialect.Oracle(), BuildSqlDialect.Oracle);
```

Comment:
Oracle will throw if snapshot is requested but in oracle readcommited == snapshot so its not a valid use case anyway see https://stackoverflow.com/a/21423042/236004